src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -6,30 +6,109 @@
 #pragma hdrstop
 #endif
 
+// Mirrors StringComparer.cs
+constexpr int StringComparer_Ordinal           = 4;
+constexpr int StringComparer_OrdinalIgnoreCase = 5;
+
 //------------------------------------------------------------------------
 // importer_vectorization.cpp
 //
 // This file is responsible for various (partial) vectorizations during import phase,
 // e.g. the following APIs are currently supported:
 //
 //   1) String.Equals(string, string)
-//   2) String.Equals(string, string, StringComparison.Ordinal)
+//   2) String.Equals(string, string, Ordinal or OrdinalIgnoreCase)
 //   3) str.Equals(string)
-//   4) str.Equals(String, StringComparison.Ordinal)
-//   5) str.StartsWith(string, StringComparison.Ordinal)
+//   4) str.Equals(String, Ordinal or OrdinalIgnoreCase)
+//   5) str.StartsWith(string, Ordinal or OrdinalIgnoreCase)
 //   6) MemoryExtensions.SequenceEqual<char>(ROS<char>, ROS<char>)
-//   7) MemoryExtensions.Equals(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//   7) MemoryExtensions.Equals(ROS<char>, ROS<char>, Ordinal or OrdinalIgnoreCase)
 //   8) MemoryExtensions.StartsWith<char>(ROS<char>, ROS<char>)
-//   9) MemoryExtensions.StartsWith(ROS<char>, ROS<char>, StringComparison.Ordinal)
+//   9) MemoryExtensions.StartsWith(ROS<char>, ROS<char>, Ordinal or OrdinalIgnoreCase)
 //
 // When one of the arguments is a constant string of a [0..32] size so we can inline
 // a vectorized comparison against it using SWAR or SIMD techniques (e.g. via two V256 vectors)
 //
-// We might add these in future:
-//   1) OrdinalIgnoreCase for everything above
-//   2) Span.CopyTo
-//   3) Spans/Arrays of bytes (e.g. UTF8) against a constant RVA data
+
+//------------------------------------------------------------------------
+// ConvertToLowerCase: Converts input ASCII data to lower case
+//
+// Arguments:
+//    input  - Constant data to change casing to lower
+//    mask   - Mask to apply to non-constant data, e.g.:
+//       input: [  h ][  i ][  4 ][  - ][  A ]
+//       mask:  [0x20][0x20][ 0x0][ 0x0][0x20]
+//    length - Length of input
+//
+// Return Value:
+//    false if input contains non-ASCII chars
 //
+static bool ConvertToLowerCase(WCHAR* input, WCHAR* mask, int length)
+{
+    for (int i = 0; i < length; i++)
+    {
+        auto ch = (USHORT)input[i];
+        if (ch > 127)
+        {
+            JITDUMP("Constant data contains non-ASCII char(s), give up.\n");
```

Comment:
@EgorBo, just an FYI, if (in the future) we wish to extend it, we can look into coreclr/pal which provides encoding-aware APIs from: https://github.com/dotnet/runtime/blob/a4b8893/src/coreclr/pal/src/locale/unicode.cpp (which, AFAIK, eventually uses same ICU bindings same as System.Globalization.Native).

Reply:
Thanks, I'm not sure we can extend it further, maybe for Latin1 range? (0...256)

I was thinking of supporting the common "culture-dependent" case with a quick "IsAscii or fallback" kind of check, but not as port of this PR and I guess I still don't need ICU for that.